src/service/navigation.js

```diff
@@ -284,13 +284,10 @@ export class Navigation {
     }
 
     // Handle anchor transformations.
-    let am = this.anchorMutators_.peek();
-    while (am) {
-      this.anchorMutators_.dequeue();
-      am(target);
+    this.anchorMutators_.forEach(anchorMutator => {
+      anchorMutator(target);
       location = this.parseUrl_(target.href);
```

Comment:
Is the location parsing necessary after every step?

Reply:
i do it because the location could have been mutated in the callback and i'd like to pass that modified location in the handleNavClick_ call below.